packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts

```diff
@@ -243,6 +243,11 @@ export class DockerImageAsset extends Construct implements IAsset {
    */
   public readonly assetHash: string;
 
+  /**
+   * The tag of this asset when it is uploaded to ECR. The tag may differ from the assetHash if a stack synthesizer adds a dockerTagPrefix.
+   */
+  public readonly synthesizedTag: string;
```

Comment:
I'm not sure about the name `synthesized`Tag. What about `imageTag` instead?

Reply:
I chose `synthesizedTag` to make it clear that this will differ from `assetHash` when a synthesizer adds a `dockerTagPrefix`. I was also worried that a user would see `imageUri` and `imageTag` and think that they are distinct instead of knowing `imageUri` includes the tag (I have done this in my project).

If you feel those concerns are relatively minor, I am fine with `imageTag`.

Reply:
Is this a public facing field or something that we use in our internal logic?

Reply:
This is a public facing field. I changed it to `imageTag` everywhere.

Reply:
`imageTag` is good. The relationship between uri and tag is docker domain knowledge. I don't think it would be a good idea to abstract that away.